desktop/lib.go

```diff
@@ -122,7 +127,7 @@ func start() {
 	}()
 
 	golog.SetPrepender(logging.Timestamped)
-	handleSignals(a)
+	// handleSignals(a)
```

Comment:
I don't think this code to handle signals is the problem area since it's been around and hasn't caused the app to crash previously. It's likely some code that changed recently..

For instance, I wonder if we maybe introduced a bug somewhere fetching and storing user data.. the fact we see this error consistently in the logs suggests an issue where a thread may be attempting to relock a mutex (like [this one](https://github.com/getlantern/lantern-client/blob/main/desktop/app/pro.go#L19) without first unlocking it:

```
goroutine 283 gp=0x14000ba3a40 m=nil [sync.Mutex.Lock]:
```

have you tried commenting out any of the new code or calls that are being made to see if the app still crashes with those disabled?

Reply:
Have we checked if the app crashes on main? Let me do that now..

Reply:
Ok, this is interesting: the app is crashing for me on main, but it no longer does if I go back to commit 042186e7d2510b7185ac3398b5284535d4cb0274. So I'm pretty sure the crash is unrelated to the changes here.

Given the crash is unrelated, let's revert the changes to disable signal handling, continue using `ParallelForIdempotent`, and merge this? Then we can debug the issue further on main.

Reply:
The error you are seeing is different then mine, this looks like some nil pointer issue, and mine is something with signal, you can check slack for more info.

Reply:
Sure, I have re-enbale signal and added proxied.ParallelForIdempotent

Reply:
I am fine with merging.

Reply:
To me it looks like this might be crash, Since this actually throws painc.
```
runtime stack:
runtime.throw({0x15c9fc20e?, 0x0?})
	runtime/panic.go:1023 +0x40 fp=0x14000051490 sp=0x14000051460 pc=0x15bce13a0
runtime.sigNotOnStack(0x10, 0x14000051538, 0x14000080008)
	runtime/signal_unix.go:1065 +0x118 fp=0x140000514c0 sp=0x14000051490 pc=0x15bcfbe08
runtime.adjustSignalStack(0x10, 0x14000080008, 0x14000051568)
	runtime/signal_unix.go:592 +0x25c fp=0x14000051530 sp=0x140000514c0 pc=0x15bcfac6c
runtime.sigtrampgo(0x10, 0x140000516d0, 0x14000051738)
	runtime/signal_unix.go:480 +0x8c fp=0x140000515b0 sp=0x14000051530 pc=0x15bcfa7ac
runtime.sigtrampgo(0x10, 0x140000516d0, 0x14000051738)
	<autogenerated>:1 +0x1c fp=0x140000515e0 sp=0x140000515b0 pc=0x15bd20adc
runtime.sigtramp()
	runtime/sys_darwin_arm64.s:227 +0x4c fp=0x140000516a0 sp=0x140000515e0 pc=0x15bd1f5ac
```

Reply:
Here is ticket-:https://github.com/getlantern/engineering/issues/1504

Reply:
Thanks, @jigar-f! I will take a closer look at this over the weekend